src/backend/columnar/columnar_storage.c

```diff
@@ -610,8 +614,16 @@ ColumnarMetapageRead(Relation rel, bool force)
 	BlockNumber nblocks = smgrnblocks(rel->rd_smgr, MAIN_FORKNUM);
 	if (nblocks == 0)
 	{
-		elog(ERROR, "columnar metapage for relation \"%s\" does not exist",
-			 RelationGetRelationName(rel));
+		/*
+		 * We only expect this to happen during binary ugrades. This is because,
+		 * in current version of columnar, we immediately create the metapage
+		 * for columnar tables, i.e right after creating the table.
+		 * However in older versions, we were creating metapages lazily, i.e
+		 * when ingesting data to columnar table.
```

Comment:
By "binary upgrades" I think you mean replacing the .so, right?

In postgres terminology, binary upgrade refers to a major version upgrade of the server.

Reply:
yes, let me fix that in this pr since it's a small change